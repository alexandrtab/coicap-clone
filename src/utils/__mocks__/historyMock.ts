export const historyDataMock = [
	{ "date": "2021-08-16T00:00:00.000Z",
		"priceUsd": "46953.4801316103380567",
		"time": "1629072000000" },

	{ "date": "2021-08-17T00:00:00.000Z",
		"priceUsd": "45986.5228431102022226",
		"time": "1629158400000" },

	{ "date": "2021-08-18T00:00:00.000Z",
		"priceUsd": "45154.6886697612712484",
		"time": "1629244800000" },

	{ "date": "2021-08-19T00:00:00.000Z",
		"priceUsd": "45226.9076561575262501",
		"time": "1629331200000" },
	{
		"date": "2021-08-20T00:00:00.000Z",
		"priceUsd": "47792.0212083321042055",
		"time": "1629417600000" },
];
