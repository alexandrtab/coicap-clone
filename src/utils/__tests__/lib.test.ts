import { historyDataMock } from "../__mocks__/historyMock";
import { handleGenerateCurrencyHistoryData } from "../lib";

describe("Generating history data", () => {
	test("should return not empty data", () => {
		const filteredData = handleGenerateCurrencyHistoryData(historyDataMock);

		expect(filteredData).not.toBeNull();
	});
	test("should filter callback be called", () => {
		const filterHistoryCallbackMock = jest.fn(history => handleGenerateCurrencyHistoryData(history));

		filterHistoryCallbackMock(historyDataMock);

		expect(filterHistoryCallbackMock.mock.calls.length).toBe(1);
	});
	test("should data be equal to snapshot", () => {
		const filteredData = handleGenerateCurrencyHistoryData(historyDataMock);

		expect(filteredData).toMatchSnapshot();
	});
});
