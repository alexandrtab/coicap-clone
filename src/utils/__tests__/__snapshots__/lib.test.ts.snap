// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generating history data should data be equal to snapshot 1`] = `
Array [
  Object {
    "date": "2021-08-16T00:00:00.000Z",
    "priceUsd": "46953.4801316103380567",
    "time": "16-7-2021",
  },
  Object {
    "date": "2021-08-17T00:00:00.000Z",
    "priceUsd": "45986.5228431102022226",
    "time": "17-7-2021",
  },
  Object {
    "date": "2021-08-18T00:00:00.000Z",
    "priceUsd": "45154.6886697612712484",
    "time": "18-7-2021",
  },
  Object {
    "date": "2021-08-19T00:00:00.000Z",
    "priceUsd": "45226.9076561575262501",
    "time": "19-7-2021",
  },
  Object {
    "date": "2021-08-20T00:00:00.000Z",
    "priceUsd": "47792.0212083321042055",
    "time": "20-7-2021",
  },
]
`;
